pkg/aws/ocpgwdeployer.go

```diff
@@ -83,14 +86,18 @@ func (d *ocpGatewayDeployer) Deploy(input api.GatewayDeployInput, status reporte
 
 	status.Success(messageValidatedPrerequisites)
 
-	status.Start("Creating Submariner gateway security group")
+	gatewaySG := ""
 
-	gatewaySG, err := d.aws.createGatewaySG(vpcID, input.PublicPorts)
-	if err != nil {
-		return status.Error(err, "unable to create gateway")
-	}
+	if !d.useLoadBalancer {
+		status.Start("Creating Submariner gateway security group")
 
-	status.Success("Created Submariner gateway security group %s", gatewaySG)
+		gatewaySG, err = d.aws.createGatewaySG(vpcID, input.PublicPorts)
+		if err != nil {
+			return status.Error(err, "unable to create gateway")
+		}
+
+		status.Success("Created Submariner gateway security group %s", gatewaySG)
+	}
 
 	subnets, err := d.aws.getSubnetsSupportingInstanceType(publicSubnets, d.instanceType)
```

Comment:
You're not actually using the subnet in the deployed machineset (which is fine), so just tagging them for submariner but not actually connecting to them makes no sense..

In fact, you need to just deploy gateway per AZ, as many as requested, so the code path is quite different.

Reply:
That's correct. so currently (non LB )we have :
- Create flow:
  1. need to find an appropriate subnet  with some logic related to AZ 
  2. tag this subnet 
  3. Create public SG
  4. Create machineset manifest (including public subnet + public SG)

- Delete flow:
 1. Get tagged subnet 
 2. Extract AZ from tagged subnet
 3. Delete machineset 
 
IIUC, tagging a subnet is something like annotation, so I thought instead of having 2 separate flows for create and delete , maintain a single flow (as possible) with some changes, for LB:

Create:
same flow, with the following changes:
Skip step 3 
and update step 4 (create machineset without SG and public subnet)

Delete:
Same flow (retrieve the AZ from subnet)

If you think that tagging a subnet for LB case is not acceptable then we need to come up with another approach, but if it's something equivalent to annotation maybe it should be OK?

Reply:
Generally a tag in AWS is just metadata so it won't affect the functionality, that is true.

However, the problem is that the amount of available public subnets could be lower that the requested amount of gateways, so to properly support LB mode it's best to avoid relying on public subnets altogether, since we don't actually need them.

Unfortunately, this would necessitate a separate code path for LB, but IMHO it's better than relying on subnets, which in an LB case is an unnecessary constraint.